x/gov/msgs.go

```diff
@@ -116,9 +94,6 @@ func (msg MsgDeposit) ValidateBasic() sdk.Error {
 	if msg.Amount.IsAnyNegative() {
 		return sdk.ErrInvalidCoins(msg.Amount.String())
 	}
-	if msg.ProposalID < 0 {
```

Comment:
I don't understand this part, `msg.ProposalID` is uint64, how can it be less than 0? Also, I'm not sure why we want to restrict the `msg.ProposalID` to be bigger than the `startingProposalID`, it will be rejected anyway. `startingProposalID` have to be a hardcoded const(because `ValidateBasic` is pure) and it makes the test broken